api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -532,7 +533,14 @@ public void setVisit(Visit visit) {
 		
 		return encounterProviders.stream()
 				.filter(ep -> includeVoided || !ep.getVoided())
-				.collect(Collectors.groupingBy(EncounterProvider::getEncounterRole, Collectors.mapping(EncounterProvider::getProvider, Collectors.toSet())));
+				.collect(Collectors.toMap((ep) -> (ep).getEncounterRole(),
```

Comment:
you seem to have stumbled on a java bug https://bugs.openjdk.java.net/browse/JDK-8148463 :)

I feel that since we cant do anything about the bug we should not use streams here since its hard to read. but thats just IMHO

what about sticking to the traditional kind:

```java
		for (EncounterProvider ep : encounterProviders) {
			if (includeVoided || !ep.getVoided()) {
				if (!result.containsKey(ep.getEncounterRole())) {
					result.put(ep.getEncounterRole(), new HashSet<>());
				}
				result.get(ep.getEncounterRole()).add(ep.getProvider());
			}
		}
		return result;
```

maybe others @dkayiwa or @wluyima or our master of streams @lluismf want to comment as well